application/cms/models.py

```diff
@@ -190,6 +190,34 @@ class DataSource(db.Model, CopyableModel):
         "MeasureVersion", secondary="data_source_in_measure_version", back_populates="data_sources"
     )
 
+    def merge(self, data_source_ids=[]):
+
+        if self.id in data_source_ids:
+            raise ValueError("Can’t merge with self")
+
+        if len(data_source_ids) == 0:
+            raise ValueError("No data source IDs specified")
+
+        data_sources_to_delete = DataSource.query.filter(DataSource.id.in_(data_source_ids))
+
+        if (data_sources_to_delete.count()) != len(data_source_ids):
+
+            difference = len(data_source_ids) - data_sources_to_delete.count()
+
+            raise ValueError(f"#{difference} data sources not found")
+
+        for data_source_to_delete in data_sources_to_delete:
+
+            DataSourceInMeasureVersion.query.filter(
+                DataSourceInMeasureVersion.data_source_id == data_source_to_delete.id
+            ).update({"data_source_id": self.id})
+
+            db.session.delete(data_source_to_delete)
+
+        db.session.commit()
```

Comment:
I know we do this all over the place, but my feeling is that `db.session.commit()` should always be in the view. We don't want individual model methods to be committing to the database - there should only be a single point when all the database changes are committed, which is immediately before returning the response. That way we know we're only making atomic changes that are committed if the entire view has succeeded.